commit edd2126aa8aab8a87db7cc480d5047e9280d7acf
Author: Jiri Kosina <jikos@kernel.org>
Date:   Thu Nov 18 16:28:43 2010 +0100

    HID: wacom: make sysfs permissions more strict
    
    It's not really dangerous in this driver, but it's against general
    practice and worth fixing.
    
    Wacom uses the attribute for changing the reporting speed of the
    tablet (and this actually requires poking the device in the
    background) (still I wouldn't consider it a security issue though).
    
    udev is a proper place to handle this.
    
    Reported-by: Linus Torvalds <torvalds@linux-foundation.org>
    Signed-off-by: Jiri Kosina <jkosina@suse.cz>

diff --git a/drivers/hid/hid-wacom.c b/drivers/hid/hid-wacom.c
index 724f46ed612f..94caae731381 100644
--- a/drivers/hid/hid-wacom.c
+++ b/drivers/hid/hid-wacom.c
@@ -172,7 +172,7 @@ static ssize_t wacom_store_speed(struct device *dev,
 		return -EINVAL;
 }
 
-static DEVICE_ATTR(speed, S_IRUGO | S_IWUGO,
+static DEVICE_ATTR(speed, S_IRUGO | S_IWUSR | S_IWGRP,
 		wacom_show_speed, wacom_store_speed);
 
 static int wacom_raw_event(struct hid_device *hdev, struct hid_report *report,